cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/AppFabricTestBase.java

```diff
@@ -404,9 +405,20 @@ protected HttpResponse deploy(Class<?> application, @Nullable String apiVersion,
 
   protected HttpResponse deploy(Id.Application appId,
                                 AppRequest<? extends Config> appRequest) throws Exception {
-    HttpEntityEnclosingRequestBase request;
     String deployPath = getVersionedAPIPath("apps/" + appId.getId(), appId.getNamespaceId());
-    request = getPut(deployPath);
+    HttpEntityEnclosingRequestBase request = getPut(deployPath);
+    request.setHeader(Constants.Gateway.API_KEY, "api-key-example");
+    request.setHeader(HttpHeaders.Names.CONTENT_TYPE, MediaType.APPLICATION_JSON);
+    request.setEntity(new StringEntity(GSON.toJson(appRequest)));
+    return execute(request);
+  }
+
+  protected HttpResponse deploy(ApplicationId appId,
```

Comment:
Since this method and the above method have the same functionality, one can call another, right? This will reduce code duplication.